### ISSUE TITLE
Ignoring dependencies

### ISSUE THREAD

**Issue by npepinpe** [revapi/revapi#244]
We have a use case where we would like to disable checking dependencies, as we have two modules which are independently checked, and one depends on the other. To avoid having to specify allowed differences/exclusions twice, we thought to disable `checkDependencies`. We've also set the Java plugin to ignore missing classes. However, while the build is successful, it produces a ton of error logs, all `NullPointerException` relating to the "missing" classes.

Is there something additional that needs to be configured, or this is the expected behavior?

We use the latest Maven plugin. Configuration is:

```json
[
  {
    "extension": "revapi.java",
    "configuration": {
      "reportUsesFor": "all-differences",
      "missing-classes": {
        "behavior": "ignore",
        "ignoreMissingAnnotations": true
      },
      "matchOverloads": false
    }
  }
]
```

And the plugin is configured with `<checkDependencies>false</checkDependencies>`

Example exception:

```
[ERROR] Failed to scan class io.grpc.stub.StreamObserver. Analysis results may be skewed.
java.lang.NullPointerException
    at org.revapi.java.matcher.PackageMatcher$1$1.getPackage (PackageMatcher.java:69)
    at org.revapi.java.matcher.PackageMatcher$1$1.doStart (PackageMatcher.java:60)
    at org.revapi.java.matcher.PackageMatcher$1$1.doStart (PackageMatcher.java:51)
    at org.revapi.base.IndependentTreeFilter.start (IndependentTreeFilter.java:38)
    at org.revapi.TreeFilter$2.lambda$start$0 (TreeFilter.java:85)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:195)
    at java.util.ArrayList$ArrayListSpliterator.forEachRemaining (ArrayList.java:1655)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:484)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:474)
    at java.util.stream.ReduceOps$ReduceOp.evaluateSequential (ReduceOps.java:913)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.reduce (ReferencePipeline.java:558)
    at org.revapi.TreeFilter$2.start (TreeFilter.java:85)
    at org.revapi.base.IncludeExcludeTreeFilter.doStart (IncludeExcludeTreeFilter.java:66)
    at org.revapi.base.IncludeExcludeTreeFilter.start (IncludeExcludeTreeFilter.java:61)
    at org.revapi.basic.ConfigurableElementFilter$1.start (ConfigurableElementFilter.java:131)
    at org.revapi.TreeFilter$2.lambda$start$0 (TreeFilter.java:85)
```

**Comment by metlos** [comment]
What versions of grpc are used by your old and new API? Also, could you please run the maven build with `-X` and share the full effective analysis configuration of  revapi, which the `revapi-maven-plugin` should output as a debug log message?

**Comment by npepinpe** [comment]
The previous gRPC version was 1.38.1, and the current version is 1.40.1.

I can guarantee `io.grpc.stub.StreamObserver` is present in both versions, if this is what you were going for :slightly_smiling_face: (see [1.38.1](https://javadoc.io/static/io.grpc/grpc-stub/1.38.1/io/grpc/stub/StreamObserver.html) and [1.40.1](https://javadoc.io/static/io.grpc/grpc-stub/1.40.1/io/grpc/stub/StreamObserver.html))

Here's the logs (as a gist since it's quite large): https://gist.github.com/npepinpe/50553353b3e707e57672a19944ecf352

This isn't a huge problem for us, the current workaround is to exclude all archives but the one under analysis and this works fine, but needs to specified every time. I would have expected `checkDependencies` to solve this in a more succinct way, but maybe I misunderstood the goal of the setting.

**Comment by metlos** [comment]
I've just released `revapi-java:0.25.0` that should fix the issue. If you wan to be Java 17 compatible, please also upgrade to the latest maven plugin - 0.14.4.

**Comment by metlos** [comment]
This should work even better in `revapi-java:0.26.0` 😄
